### PR TITLE
feat(ddm): Add quota for disabling custom metrics for a specific org or project

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds a killswitch for custom metrics and also adds a new abuse quota for throttling the limit of custom metrics per org.

### CODE REVIEW

File: src/sentry/quotas/base.py [hunk 1]
```diff
@@ -491,12 +498,12 @@ def get_abuse_quotas(self, org):
             # Negative limits in config mean a reject-all quota.
             if limit < 0:
                 yield QuotaConfig(
+                    limit=0,
                     scope=quota.scope,
                     categories=quota.categories,
-                    limit=0,
                     reason_code="disabled",
+                    namespace=quota.namespace,
```
Comment (iambriccardo) [inline src/sentry/quotas/base.py:505]: Added this since it was missing.

File: src/sentry/quotas/base.py [hunk 2]
```diff
@@ -419,6 +419,13 @@ def get_abuse_quotas(self, org):
                 categories=[DataCategory.METRIC_BUCKET],
                 scope=QuotaScope.ORGANIZATION,
             ),
+            AbuseQuota(
```
Comment (iambriccardo) [inline src/sentry/quotas/base.py:422]: Added this abuse quota to have additional control of the custom namespace in the future. This is not tied to the killswitch.

File: src/sentry/quotas/redis.py [hunk 3]
```diff
@@ -67,6 +68,20 @@ def get_quotas(
 
         results = [*self.get_abuse_quotas(project.organization)]
 
+        # If the organization belongs to the killswitched list, we want to stop ingesting custom metrics.
+        if project.organization.id in (
```
Comment (iambriccardo) [inline src/sentry/quotas/redis.py:72]: I opted to give this killswitch an explicit implementation here but in the future we could just rely on the abuse quota defined above.

File: src/sentry/options/defaults.py [hunk 4]
```diff
@@ -1814,6 +1821,13 @@
     flags=FLAG_AUTOMATOR_MODIFIABLE,
 )
 
+# IDs of orgs that will stop ingesting custom metrics.
+register(
+    "custom-metrics-ingestion-killswitched-orgs",
```
Comment (jan-auer) [inline src/sentry/options/defaults.py:1826]: Let's also add a way to disable specific project, as per Armin's request.

**Review by jan-auer — APPROVED** [review]

**Pull request merged**